src/index.ts

```diff
@@ -0,0 +1,70 @@
+import express from "express";
+import bodyParser from "body-parser";
+import { createClient } from "redis";
+import "./config";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+import { scheduleJobs } from "./jobs.js";
+import {
+  syncServers,
+  syncHashes,
+  storeMessage,
+  searchByRecipient,
+  searchByHash,
+} from "./util.js";
+
+const app = express();
+app.use(bodyParser.json());
+
+export const redis = createClient({
+  url: process.env.REDIS_URL,
+});
+
+redis.on("error", (err) => console.error("Redis Client Error", err));
+await redis.connect();
+
+scheduleJobs();
+
+// TODO: more error handling and HTTP response codes
+
+app.post("/fetch", async (req, res) => {
+  try {
+    if (req.body.recipient) {
+      console.log("Fetching messages for recipient", req.body.recipient);
+      const messages = await searchByRecipient(req.body.recipient);
+      res.json({ messages });
+    } else if (req.body.hashes) {
+      console.log("Fetching messages by hashes", req.body.hashes);
+      const messages = await Promise.all(req.body.hashes.map(searchByHash));
+      res.json({ messages });
+    } else {
+      res.status(400).send("Bad Request");
+    }
+  } catch (error) {
+    console.error("Error in /fetch endpoint:", error);
+    res
+      .status(500)
+      .json({ error: "An error occurred while fetching messages" });
+  }
+});
+
+app.post("/send", (req, res) => {
+  console.log("Storing message", req.body);
+  storeMessage(req.body);
+  res.status(200).send("ok");
```

Comment:
This is redundant, 200 is the code for ok

```suggestion
  res.status(200);
```

Reply:
I think we wanna send the actual response so...